### PR TITLE
Refactor/job details

### PR DESCRIPTION
**Pull request by AbuHussien28**
1 - refactor view model
2- set custom view one for Expandable card and details_chip in fragment about job
3 - make logic for RepositoryImp
video task  https://github.com/The-Cupcake-team/Jobs-finder/assets/96633008/af7ed385-190a-40c3-9300-c8f85a21be6d
Please review and provide feedback. Thank you! ❤️

### CODE REVIEW

File: data/repository/src/main/java/com/cupcake/repository/mapper/JobMapper.kt [hunk 1]
```diff
@@ -1,48 +1,18 @@
...
-//        jobSalary = jobSalary
-//    )
-//}
\ No newline at end of file
+package com.cupcake.repository.mapper
+
+import com.cupcake.models.Job
+import com.cupcake.remote.response.job.JobDto
+
+fun JobDto.toJob(): Job {
```
Comment (AmeerAmjed) [inline data/repository/src/main/java/com/cupcake/repository/mapper/JobMapper.kt:52]: why default value same this

Comment (Hasa9nAli) [reply data/repository/src/main/java/com/cupcake/repository/mapper/JobMapper.kt:52]: what ?

**Review by AmeerAmjed — CHANGES_REQUESTED** [review]
implementation design have issue 🤮 need more focus , not every thing in design is implement ??? why?

**Comment by AmeerAmjed** [comment]
image

**Pull request merged**